packages/libs/eda/src/lib/core/components/computations/plugins/differentialabundance.tsx

```diff
@@ -0,0 +1,273 @@
+import { useMemo } from 'react';
+import { ComputationStepContainer } from '../ComputationStepContainer';
+import { sharedConfigCssStyles } from './abundance';
+import VariableTreeDropdown from '../../variableTrees/VariableTreeDropdown';
+import { ValuePicker } from '../../visualizations/implementations/ValuePicker';
+import { useToggleStarredVariable } from '../../../hooks/starredVariables';
+import { config } from 'process';
+
+/**
+ * Differential abundance
+ *
+ * The differential abundance app is used to find taxa, genes, pathways, etc. that
+ * are more abundant in one group of samples than another. This app takes in a continuous variable
+ * collection (for example, abundance of 100 Species for all samples) as well as a way to split the samples
+ * into two groups (for example, red hair and green hair). The computation then returns information on how
+ * differentially abundance each item (taxon, gene, etc.) is between the two groups. See VolcanoPlotDataPoint
+ * for more details. Importantly, the returned data lives outside the variable tree because each returned
+ * data point corresponds to an item (taxon, gene, etc.).
+ *
+ * Currently the differential abundance app will be implemented with only a volcano visualization. Plans for
+ * the future of this app include a lefse diagram, tables of results, adding new computation methods, and
+ * strategies to create user-defined collections from the output of the computation.
+ */
+
+export type DifferentialAbundanceConfig = t.TypeOf<
+  typeof DifferentialAbundanceConfig
+>;
+
+const Comparator = t.type({
+  variable: VariableDescriptor,
+  groupA: t.array(t.string),
+  groupB: t.array(t.string),
+});
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export const DifferentialAbundanceConfig = t.type({
+  collectionVariable: VariableDescriptor,
+  comparator: Comparator,
+  differentialAbundanceMethod: t.string,
+});
+
+export const plugin: ComputationPlugin = {
+  configurationComponent: DifferentialAbundanceConfiguration,
+  configurationDescriptionComponent:
+    DifferentialAbundanceConfigDescriptionComponent,
+  createDefaultConfiguration: () => undefined,
+  isConfigurationValid: DifferentialAbundanceConfig.is,
+  visualizationPlugins: {
+    volcanoplot: volcanoPlotVisualization, // Must match name in data service and in visualization.tsx
+  },
+};
+
+function DifferentialAbundanceConfigDescriptionComponent({
+  computation,
+}: {
+  computation: Computation;
+}) {
+  const studyMetadata = useStudyMetadata();
+  const collections = useCollectionVariables(studyMetadata.rootEntity);
+  assertComputationWithConfig<DifferentialAbundanceConfig>(
+    computation,
+    Computation
+  );
+  const { configuration } = computation.descriptor;
+  const collectionVariable =
+    'collectionVariable' in configuration
+      ? configuration.collectionVariable
+      : undefined;
+  const comparatorVariable =
+    'comparatorVariable' in configuration
```

Comment:
I noticed the description was always rendering "Not selected" and its due to this line. We should be checking for `'comparator' in configuration`. Then, we'll also need to get the variable's display name so we're not just displaying the `variableId` further down, which won't mean anything to our users! Could be done with the `useFindEntityAndVariable` hook, but there may be something else out there to do so.